File: bmi.cs
using System;

namespace BMI
{
    class Program
    {
        static void Main(string[] args)
        {
            if (args.Length != 2)
            {
                Console.WriteLine("Usage: bmi <weight [kg]> <height [m]>", args[0]);
                Environment.Exit(1);
            }

            double weight = Double.Parse(args[0]);
            double height = Double.Parse(args[1]);

            double bmi = weight / Math.Pow(height, 2);

            Console.WriteLine("Your BMI is: {0}", bmi);

            if (bmi <= 18.5)
            {
                Console.WriteLine("You're underweight.");
            }
            else if (bmi <= 25)
            {
                Console.WriteLine("You're healthy.");
            }
            else if (bmi <= 30)
            {
                Console.WriteLine("You're overweight.");
            }
            else
            {
                Console.WriteLine("Holy McShizzle, you're freaking whale!");
            }
        }
    }
}

File: bmi.rs
use std::env;
use std::error::Error;

fn main() -> Result<(), Box<dyn Error>> {
    let args: Vec<_> = env::args().collect();
    if args.len() != 3 {
        println!("Usage: {} <weight [kg]> <height [m]>", args[0]);
        return Ok(());
    }

    let weight = args[1].parse::<f64>()?;
    let height = args[2].parse::<f64>()?;
    let bmi = weight / height.powi(2);
    println!("Your BMI is: {}", bmi);

    println!("{}", match bmi {
        bmi if bmi <= 18.5 => "You're underweight.",
        bmi if bmi <= 25.0 => "You're healthy.",
        bmi if bmi <= 30.0 => "You're overweight.",
        _ => "Holy McShizzle, you're freaking whale!"
    });

    Ok(())
}
